core/src/main/java/io/grpc/internal/ConnectivityStateManager.java

```diff
@@ -63,7 +63,13 @@ void notifyWhenStateChanged(Runnable callback, Executor executor, ConnectivitySt
    */
   void gotoState(@Nonnull ConnectivityState newState) {
     checkNotNull(newState, "newState");
-    checkState(state != null, "ConnectivityStateManager is already disabled");
+    if (newState != ConnectivityState.SHUTDOWN) {
+      checkState(state != null, "ConnectivityStateManager is already disabled");
+    } else if (state == null ) {
+      // When ConnectivityStateManager is already disabled, then channel shutdown is called.
+      // Keep state being null.
+      return;
+    }
```

Comment:
ConnectivityStateManager should not discriminate states based on the assumption of who calls `gotoState()`, because the manager may be used in other cases where the assumption may not hold.
I think the issue is that `ManagedChannelImpl.shutdown()` calls `gotoState()` when the manager is already disabled. To fix this immediate issue, we can add an `isDisabled()` to the manager and avoid calling `gotoState()` if it returns `true`.